### PR TITLE
add docs, make couple bugfixes

### PR DESCRIPTION
**Pull request by shahzzzam**
**Purpose of the PR:**
- Adds more documentation for custom credentials.
- Rename `armResourceId` to `aadResourceId`.
- Remove `az-cloud-name`

**Fixes #**
- Fix a bug where it couldn't make POST request using `client.Do`. So using `client.Post` instead.

### CODE REVIEW

File: docs/custom_registry_login.md [hunk 1]
```diff
@@ -0,0 +1,218 @@
+# Custom registry login using Identity
```
Comment (sajayantony) [inline docs/custom_registry_login.md]: Generic feedback. Start sentences with Upper case.

File: docs/custom_registry_login.md [hunk 2]
```diff
@@ -0,0 +1,218 @@
...
+### Create Azure VM with identity
+
+```
+az vm create -n testacbvm -g myResourceGroup --image UbuntuLTS --assign-identity $ID
+```
+
+### SSH to VM and set up with docker and acb image
+
+```
+sudo -s
```
Comment (sajayantony) [inline docs/custom_registry_login.md]: As a best practice try not dropping to root and just run `sudo apt-get install docker-ce -y`

File: graph/registry_credential.go [hunk 3]
```diff
@@ -15,7 +15,7 @@ var (
 	errInvalidUsername      = errors.New("username can't be empty")
 	errInvalidPassword      = errors.New("password can't be empty")
 	errInvalidIdentity      = errors.New("identity can't be empty")
-	errInvalidArmResourceID = errors.New("armResourceId can't be empty")
```
Comment (sajayantony) [inline graph/registry_credential.go]: Recommend renaming this to `resourceId` ref  https://github.com/Azure/azure-sdk-for-go/blob/1951233eb944a49aa5f7278cba8e3e32a8c958af/services/resources/mgmt/2018-05-01/resources/resources.go#L137

File: graph/registry_credential_test.go [hunk 4]
```diff
@@ -33,7 +33,7 @@ func TestCreateCredentialFromString(t *testing.T) {
 		{`{"registry":"r","identity":"clientID", "armResourceId": "https://management.azure.com"}`, true, &RegistryCredential{
 			Registry:      "r",
 			Identity:      "clientID",
-			ArmResourceID: "https://management.azure.com",
+			ArmResource: "https://management.azure.com",
```
Comment (sajayantony) [inline graph/registry_credential_test.go]: Could we just use the azure SDK DefaultBaseURI definition - https://github.com/Azure/azure-sdk-for-go/blob/master/services/resources/mgmt/2018-05-01/resources/client.go#L29

Comment (northtyphoon) [reply graph/registry_credential_test.go]: Strictly speaking, baseuri is different from resource registered in AAD. For most resources, they might be same but arm is a special case.
Here, the string is used for test and can be anything. It doesn't really matter.

File: graph/registry_credential_test.go [hunk 5]
```diff
@@ -33,7 +33,7 @@ func TestCreateCredentialFromString(t *testing.T) {
 		{`{"registry":"r","identity":"clientID", "armResourceId": "https://management.azure.com"}`, true, &RegistryCredential{
```
Comment (northtyphoon) [inline graph/registry_credential_test.go]: still `armResourceId`?

File: graph/registry_credential.go [hunk 6]
```diff
@@ -93,13 +93,13 @@ func CreateRegistryCredentialFromString(str string) (*RegistryCredential, error)
...
 		}
-		if cred.ArmResourceID == "" {
+		if cred.ArmResource == "" {
 			return nil, errInvalidArmResourceID
 		}
 		retVal = &RegistryCredential{
 			Registry:      cred.Registry,
 			Identity:      cred.Identity,
-			ArmResourceID: cred.ArmResourceID,
+			ArmResource: cred.ArmResource,
```
Comment (northtyphoon) [inline graph/registry_credential.go]: nit: format

File: graph/registry_credential_test.go (hunk 4 above)
Comment (sajayantony) [reply graph/registry_credential_test.go]: The property at this point is not self describing. Is it’s a URI or a resourceId or an endpoint etc.

File: docs/custom_registry_login.md (hunk 2 above)
Comment (shahzzzam) [reply docs/custom_registry_login.md]: I wanted to be consistent with some of our other docs:
https://github.com/Azure/acr-builder/blob/38f06e5acb94854c74f6fb03f7958dc711dfbd90/docs/task-with-secrets.md#ssh-to-vm-and-set-up-with-docker-and-acb-image

Comment (shahzzzam) [reply docs/custom_registry_login.md]: but you're right, I updated it.

**Pull request merged**